keyform/views.py

```diff
@@ -25,11 +27,55 @@ def get_queryset(self):
 
 class RequestView(LoginRequiredMixin, UpdateView):
     model = Request
+    success_url = reverse_lazy('home')
     template_name = "keyform/request.html"
     form_class = EditForm
 
     def get_success_url(self):
-        return reverse('home')
+        return reverse_lazy('home')
+
+class ContactView(LoginRequiredMixin, TemplateView):
+    template_name = "keyform/contact.html"
+
+    def get_context_data(self):
+        context = super(ContactView, self).get_context_data()
+        context["buildings"] = Building.objects.all()
+        return context
+
+    def post(self, request, *args, **kwargs):
+
+        if request.user.has_perm('keyform.delete_contact'):
+            pk = request.POST['pk']
+            Contact.objects.filter(pk=pk).delete();
+
+            context = self.get_context_data(**kwargs)
+            return self.render_to_response(context)
+
+class EditContactView(LoginRequiredMixin, PermissionRequiredMixin, UpdateView):
+    template_name = "keyform/contact_form.html"
+    model = Contact
+    form_class = ContactForm
+    success_url = reverse_lazy('contact')
+    permission_required = 'keyform.change_contact'
+    raise_exception = True
+
+    def get_context_data(self, **kwargs):
```

Comment:
You don't need to override `get_context_data()` anymore since the `title` variable is no longer used in the template.